pkg/kubernetes/node/v1alpha1/kubernetes.go

```diff
@@ -130,6 +141,34 @@ func NewKubeClient(opts ...KubeClientBuildOption) *Kubeclient {
 	return k
 }
 
+// KubeClientInstanceOrDie returns the singleton instance of Kubeclient
+//
+// Usage:
+//  Caller code will use syntax(-es) as shown below:
+//
+// ```go
+// import (
+//  node "github.com/openebs/maya/pkg/kubernetes/node/v1alpha1"
+// )
+//
+// node.KubeClientInstanceOrDie().Get(...)
+// node.KubeClientInstanceOrDie().Create(...)
+// node.KubeClientInstanceOrDie().Update(...)
+// node.KubeClientInstanceOrDie().List(...)
+// ```
+func KubeClientInstanceOrDie(opts ...KubeClientBuildOption) *Kubeclient {
+	once.Do(func() {
+		k := NewKubeClient(opts...)
+		_, err := k.getClientsetOrCached()
+		if err != nil {
+			glog.Fatalf("failed to initialise kubeclient instance: {%v}", err)
```

Comment:
Can we do some thing like initializing `kubeclient` outside of once.Do() and return error if any?
In once.DO() can we have initializing kubeClientInst?

Reply:
I think if we want `node.KubeClientInstanceOrDie().Create(...)` kind of usability then we can ignore above comments.

Reply:
I guess `NewKubeClient` is the one that has network trips or atleast does a lot of stuff before instantiating the kube client. Hence it has to be within `once.Do`